api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/MessageProcessorScheduler.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.dhsc.htbhf.claimant.scheduler;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import net.javacrumbs.shedlock.core.SchedulerLock;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import uk.gov.dhsc.htbhf.claimant.message.MessageProcessor;
+import uk.gov.dhsc.htbhf.claimant.message.MessageType;
+import uk.gov.dhsc.htbhf.requestcontext.aop.NewRequestContextWithSessionId;
+
+/**
+ * Component responsible for scheduling the processing of messages.
+ * Each MessageType has its own cron schedule - see https://stackoverflow.com/a/26147143 for examples.
+ * Each messageType also has its own minimum and maximum lock period. For a definition of the ISO-8601 duration format, see
+ * https://www.digi.com/resources/documentation/digidocs/90001437-13/reference/r_iso_8601_duration_format.htm.
+ * Note that the {@link MessageProcessor} also specifies a limit on the number of messages to process
```

Comment:
We might also want to think about making this configurable on a message type basis too.

Reply:
True